docs/src/components/ComponentPage/ComponentPage.js

```diff
@@ -0,0 +1,186 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016-2019 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow strict */
+
+import React, { type Node } from 'react';
+import { cssModules } from 'bpk-react-utils';
+import BpkContentContainer from 'bpk-component-content-container';
+import { BpkList, BpkListItem } from 'bpk-component-list';
+import BpkLink from 'bpk-component-link';
+import BpkText, { TEXT_STYLES, WEIGHT_STYLES } from 'bpk-component-text';
+
+import BpkMarkdownRenderer from '../DocsPageBuilder/BpkMarkdownRenderer';
+import getMarkdownString from '../../helpers/markdown-helper';
+import UsageTable from '../UsageTable';
+
+import STYLES from './ComponentPage.scss';
+
+const getClassName = cssModules(STYLES);
+
+type ContentType = string | Node;
+
+type Props = {
+  examples: Array<{
+    id: string,
+    title: string,
+    blurb?: ContentType,
+    content?: ContentType,
+  }>,
+  readme: string,
+  additionalContent: ?Array<{
+    id: string,
+    title: string,
+    content: ContentType,
+  }>,
+  usageTable: ?{
+    dos: Array<string>,
+    donts: Array<string>,
+  },
+};
+
+const ComponentPage = (props: Props) => {
+  const { additionalContent, examples, readme, usageTable } = props;
+  return (
+    <div>
+      {/* Table of contents */}
```

Comment:
This has probably been done with reason but wondering instead of just having all of these in one page with comments to explain what they do, could they be broken down into components that are then included on the page just to help with future maintenance and make the file lighter?

Just an idea 😄

Reply:
Are you thinking something like this?
```
<TableOfContents name="Examples" items={examples} />
<TableOfContents name="Additional content" items={additionalContent} />
```

Reply:
I considered it, but I think it would actually **increase** maintenance. Right now it's a longish file, but the contents are very simple. You've got:

1. A few instances of `BpkList` to generate the table of contents
1. A few blocks / `array.map`s to generate the content

If there were complex logic involved in those things, then I think there'd be a better argument for splitting it, but it's really just null checks and iterators.

 Splitting brings its own maintenance burden (it would need props defining and so on) and I don't think the benefits are greater than the cost.

Reply:
No problem, with that reasoning I am happy to have this the way it is, just wanted to raise this incase it had been missed

Reply:
Yeah, always good to check these assumptions!